### PR TITLE
MSPSDS-869 update existing business pages

### PR DESCRIPTION
**Pull request by MatthewBurstein**

### CODE REVIEW

File: mspsds-web/app/helpers/businesses_helper.rb [hunk 1]
```diff
@@ -27,47 +27,35 @@ def sort_direction
...
-      :company_type_code,
-      :company_status_code,
-      :nature_of_business_id,
-      :additional_information,
-      location_attributes: %i[id address phone_number locality country postal_code _destroy]
+      :id,
+      :legal_name,
+      :trading_name,
+      :company_number,
+      locations_attributes: %i[id address_line_1 address_line_2 phone_number county country postal_code _destroy],
```
Comment (broder) [inline mspsds-web/app/helpers/businesses_helper.rb]: Is `_destroy` used anymore?

File: mspsds-web/app/helpers/businesses_helper.rb [hunk 2]
```diff
@@ -27,47 +27,35 @@ def sort_direction
...
       defaults_on_primary_location(@business)
       @business.source = UserSource.new(user: current_user)
     else
       @business = Business.new
       @business.locations.build
+      @business.contact = Contact.new
     end
   end
 
   def advanced_search(excluded_ids = [])
```
Comment (broder) [inline mspsds-web/app/helpers/businesses_helper.rb]: I think this method was used to add the suggestions on the right hand side so we won't need it anymore (and some of the others below as well).

File: mspsds-web/app/models/business.rb [hunk 3]
```diff
@@ -75,11 +47,6 @@ def pretty_description
   def add_sic_code(c_h_info)
```
Comment (broder) [inline mspsds-web/app/models/business.rb]: Do we need this method anymore?

File: mspsds-web/app/controllers/businesses_controller.rb [hunk 4]
```diff
@@ -31,18 +27,6 @@ def edit
     @business.locations.build unless @business.locations.any?
   end
 
-  # GET /businesses/suggested
-  def suggested
```
Comment (broder) [inline mspsds-web/app/controllers/businesses_controller.rb]: I think there's quite a lot of `suggested` logic in the JavaScript which we should remove as well.

File: mspsds-web/db/migrate/20190115111318_update_business_and_location_fields.rb [hunk 5]
```diff
@@ -0,0 +1,60 @@
+class UpdateBusinessAndLocationFields < ActiveRecord::Migration[5.2]
+  class Location < ApplicationRecord; end
```
[the remark this answers is not in the document]
Comment (MatthewBurstein) [reply mspsds-web/db/migrate/20190115111318_update_business_and_location_fields.rb]: I did this on Jan's advice following the Stubbing Out Models section [here](https://gist.github.com/ccschmitz/7f4df754cc8933a2033c). Happy to remove it if you feel it isn't necessary though.

[the remark this answers is not in the document]
Comment (broder) [reply mspsds-web/db/migrate/20190115111318_update_business_and_location_fields.rb]: Fair enough - seems sensible!

File: mspsds-web/app/helpers/businesses_helper.rb [hunk 6]
```diff
@@ -17,7 +17,7 @@ def search_for_businesses(page_size)
   end
 
   def sort_column
```
[the remark this answers is not in the document]
Comment (MatthewBurstein) [reply mspsds-web/app/helpers/businesses_helper.rb]: Does that mean that I don't need to take action here as part of this ticket?

[the remark this answers is not in the document]
Comment (broder) [reply mspsds-web/app/helpers/businesses_helper.rb]: You don't need to sort this here, no.

File: mspsds-web/app/helpers/businesses_helper.rb [hunk 7]
```diff
@@ -27,130 +27,29 @@ def sort_direction
...
-
-  def companies_house_new_business_form_url
-    @investigation.present? ? companies_house_investigation_businesses_path(@investigation) : companies_house_businesses_path
-  end
-
   def create_business
     if params[:business]
       @business = Business.new(business_params)
       @business.locations.build unless @business.locations.any?
+      @business.contact = Contact.new(business_params[:contact_attributes]) unless @business.contact
```
Comment (broder) [inline mspsds-web/app/helpers/businesses_helper.rb]: Just to check, Rails doesn't automatically map `contact_attributes` in `business_params` to a contact model? (i.e. does it break if we remove this line?)

Comment (MatthewBurstein) [reply mspsds-web/app/helpers/businesses_helper.rb]: Yes you're right this does seem to happen automatically. I've removed the line.

**Pull request merged**